corehq/apps/user_importer/validation.py

```diff
@@ -136,12 +138,31 @@ def validate_spec(self, spec):
 class TableauRoleValidator(ImportValidator):
     _error_message = _("Invalid tableau role: '{}'. Please choose one of the following: {}")
 
+    def __init__(self, domain):
+        super().__init__(domain)
+        self.valid_role_options = [option[1] for option in TABLEAU_ROLES]
+
     def validate_spec(self, spec):
-        valid_role_options = [option[1] for option in TABLEAU_ROLES]
         tableau_role = spec.get('tableau_role')
+        if tableau_role is not None and tableau_role not in self.valid_role_options:
+            return self._error_message.format(tableau_role, ', '.join(self.valid_role_options))
+
+
+class TableauGroupsValidator(ImportValidator):
+    _error_message = _("These groups, {}, are not valid for this domain. Please choose one of the following: {}")
 
-        if tableau_role is not None and tableau_role not in valid_role_options:
-            return self._error_message.format(tableau_role, ', '.join(valid_role_options))
+    def __init__(self, domain):
+        super().__init__(domain)
+        self.allowed_groups_for_domain = get_allowed_tableau_groups_for_domain(self.domain)
+
+    def validate_spec(self, spec):
+        tableau_groups = spec.get('tableau_groups').split(',')
```

Comment:
Do we need to worry about the `None` case here for `spec.get('tableau_groups')`?

Reply:
also a good catch. I'll check for None before splitting.

Reply:
https://github.com/dimagi/commcare-hq/commit/350af916686d0cdd531a2214a0bc9ad54737b6cc